server/src/routes/api.ts

```diff
@@ -23,55 +74,48 @@ router.post('/word/add', (req: Request, res: Response) => {
     id: new mongoose.Types.ObjectId
   });
 
-  word.save()
-    .then((word: Word) => res.json(word))
-    .catch(err => res.send(err));
-});
-
-router.put('/word/list', (req: Request, res: Response) => {
-  const updatedList = req.body.list;
+  try {
+    const saved: Word = await word.save();
 
-  if (!Array.isArray(updatedList)) {
-    return res.status(400).json({'error': 'Bad Data'});
+    res.json(saved);
+  } catch(err) {
+    res.send(err);
   }
-
-  for (let i = 0; i < updatedList.length; i++) {
-    WordModel.update({'id': updatedList[i].id}, updatedList[i])
-      .catch(err => res.send(err));
-  }
-
-  WordModel.find()
-    .then((words: Word[]) => res.json(words))
-    .catch(err => res.send(err));
 });
 
-router.delete('/word/list', (req: Request, res: Response) => {
-  const deleteList = req.body.list;
-
-  if (!Array.isArray(deleteList)) {
-    return res.status(400).json({'error': 'Bad Data'});
-  }
-
-  function removeList() {
-    return new Promise((resolve) => {
-      deleteList.forEach((id, index) => {
-        WordModel.findOne({id: id}, null, {limit: 1})
-          .then((word: Word) => {
-            word.remove();
-            if (index === deleteList.length - 1) resolve()
-          })
-          .catch(err => res.send(err));
-      });
-
-    })
-  }
-
-  removeList().then(() => {
-    WordModel.find()
-      .then((words: Word[]) => res.json(words))
-      .catch(err => res.send(err))
+router.route('/word/:id')
+  .get(async (req: Request, res: Response) => {
+    try {
+      const foundWord: Word = await WordModel.findOne({id: req.params.id});
+
+      if (!foundWord) return res.json({'message': 'No word found'});
+
+      res.json(foundWord);
+    } catch(err) {
+      res.send(err);
+    }
+  })
+  .put(async (req: Request, res: Response) => {
+    try {
+      const updated: Word = await WordModel.findOneAndUpdate({id: req.params.id}, req.body.word, {new: true});
```

Comment:
Domyślnie funkcja findOneAndUpdate, zwraca obiekt przed modyfikacją. W momencie jak zdefiniuje {new: true}, to zostaje zwrócony już zmodyfikowany obiekt